pkg/api/norman/customization/aks/listers.go

```diff
@@ -120,6 +122,94 @@ func (s sortableVersion) Less(a, b int) bool {
 	return version.Compare(s[a], s[b], "<")
 }
 
+type KubernetesUpgradeVersion struct {
+	Version string `json:"version"`
+	Enabled bool   `json:"enabled"`
+}
+
+type KubernetesUpgradeVersions []*KubernetesUpgradeVersion
+
+func (s KubernetesUpgradeVersions) Len() int {
+	return len(s)
+}
+
+func (s KubernetesUpgradeVersions) Swap(a, b int) {
+	s[a], s[b] = s[b], s[a]
+}
+
+func (s KubernetesUpgradeVersions) Less(a, b int) bool {
+	return version.Compare(s[a].Version, s[b].Version, "<")
+}
+
+// listKubernetesUpgradeVersions lists all kubernetes versions listed by AKS Container Service and marks which ones the
+//given cluster can be upgraded to.  A version's `Enabled` flag is true if the cluster can be upgraded to the version
+//in its current state.
+func listKubernetesUpgradeVersions(ctx context.Context, clusterLister v3.ClusterLister, cap *Capabilities) ([]byte, int, error) {
+	// load the target cluster, if the cluster is not found we cannot proceed
+	cluster, err := clusterLister.Get("", cap.ClusterId)
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to load cluster: %v", err)
```

Comment:
If the cluster is not found, I can get behind 400. I've seen 404 used before, but it's really somewhat of a controversial topic - using 404 for API responses.
However, if there is any other reason, it should probably be 500.